### PR TITLE
For upstream/ayaz cpu tests

### PR DESCRIPTION
**Pull request by ayaz91**

### CODE REVIEW

**Comment by powerjg** [comment]
Just a quick comment: I notice there's something wrong with the commit messages. They should all follow the guidelines: https://gem5.googlesource.com/public/gem5/+/master/CONTRIBUTING.md#requirements-for-change-descriptions

Also, make sure you're using rebase and not merge so you don't try to push merge commits upstream.

File: tests/gem5/fixture.py [hunk 1]
```diff
@@ -220,11 +220,9 @@ def __init__(self, path, program, **kwargs):
         super(DownloadedProgram, self).__init__("download-" + program,
                                                 build_once=True, **kwargs)
 
-        self.program_dir = joinpath('test-progs', path)
+        self.program_dir = path
```
Comment (powerjg) [inline tests/gem5/fixture.py]: This should probably be a separate commit.

File: tests/gem5/hello_se/test_hello_se.py [hunk 2]
```diff
@@ -39,7 +39,7 @@
 for isa in test_progs:
     for binary in test_progs[isa]:
         import os
-        path = os.path.join('hello', 'bin', isa, 'linux')
+        path = os.path.join('test-progs', 'hello', 'bin', isa, 'linux')
```
Comment (powerjg) [inline tests/gem5/hello_se/test_hello_se.py]: with this

File: tests/gem5/m5_util/test_exit.py [hunk 3]
```diff
@@ -37,7 +37,7 @@
 r'Exiting @ tick \d* because m5_exit instruction encountered'
 )
 
-test_program = DownloadedProgram('m5-exit/bin/x86/linux/', 'm5_exit')
+test_program = DownloadedProgram('test-progs/m5-exit/bin/x86/linux/', 'm5_exit')
```
Comment (powerjg) [inline tests/gem5/m5_util/test_exit.py]: and this

**Comment by powerjg** [comment]
Should this PR be closed?

**Comment by ayaz91** [comment]
Closed!
